packages/network-controller/src/NetworkController.ts

```diff
@@ -738,41 +703,6 @@ export class NetworkController extends BaseControllerV2<
     await this.lookupNetwork();
   }
 
-  /**
-   * Fetches the network ID for the network, ensuring that it is a hex string.
-   *
-   * @param networkClientId - The ID of the network client to fetch the network
-   * @returns A promise that either resolves to the network ID, or rejects with
-   * an error.
-   * @throws If the network ID of the network is not a valid hex string.
-   */
-  async #getNetworkId(networkClientId: NetworkClientId): Promise<NetworkId> {
```

Comment:
Do we want to continue to expose this method but make it public for case where, as you put it in the PR description:

> It will still be possible to get networkId using the rpc client to make a call to the net_version method for cases that truly still rely on it.

Reply:
Idk, i tried doing it that way, but I ended up liking it more with the `net_version` call pushed into extension's MetaMaskController as doing so removes networkId from NetworkController pretty much entirely. It's also going to only going to be temporarily living in extension since we can get rid of it there after we go through the deprecation lifecycle for `networkVersion`

Reply:
Ok sounds reasonable.